src/storm/environment/modelchecker/ModelCheckerEnvironment.h

```diff
@@ -24,6 +25,9 @@ class ModelCheckerEnvironment {
     SteadyStateDistributionAlgorithm getSteadyStateDistributionAlgorithm() const;
     void setSteadyStateDistributionAlgorithm(SteadyStateDistributionAlgorithm value);
 
+    ConditionalAlgorithm getConditionalAlgorithm() const;
+    void setConditionalAlgorithm(ConditionalAlgorithm value);
```

Comment:
It doesnt matter performance-wise, but do we pass enums by value or by ref?

Reply:
We've been doing this mostly by ref in the past, but imo it is not reasonable to do or even enforce doing that:

- Performance-wise, there is no practical difference. Arguably, by-ref is slightly worse as it requires 8 bytes and an additional indirection whereas `enum class`es are essentially `int`s and require 4 bytes (on standard machines).
- Semantics-wise, there is always the chance that unexpected side effects might happen. For example, the reference might get invalidated in the function body because of a vector re-allocation or whatever. Of course, not the case here but generally avoiding references (whenever they are not the intended semantics) means safer code.

Reply:
Also a good point. There's no silver bullet and I don't think we have been consistent so far.